quesma/table_resolver/table_resolver.go

```diff
@@ -281,8 +292,11 @@ func NewTableResolver(quesmaConf config.QuesmaConfiguration, discovery clickhous
 		pipelineName: IngestPipeline,
 
 		resolver: &compoundResolver{
+			patternSplitter: patternSplitter{
```

Comment:
Code is self explaining, however it would make sense to add general idea description (something you added as PR comment)

```
1) patternSplitter splits a pattern (e.g. logs*) into concrete single indexes (e.g. logs1, logs2)
2) Existing rules are evaluated on each single index separately
3) decisionMerger merges decisions of single indexes, making sure that the decisions are compatible. It yields a single decision.
```